src/pages/signup.jsx

```diff
@@ -1,114 +1,146 @@
 import Image from "next/image";
 import CALogo from "public/ca-logo.svg";
 import GreenWaves from "@/components/GreenWaves";
+import GoogleLogo from "public/google-logo.svg";
 import Link from "next/link";
 import { signIn } from "next-auth/react";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { signUpSchema } from "@/utils/consts";
+import { ExclamationCircleIcon } from '@heroicons/react/24/solid';
+import toast from "react-hot-toast";
 
 /**
  * Sign up page
  *
  * @returns {React.ReactElement} The sign up page
  */
 export default function Signup({ dogs }) {
-  const onSubmitForm = async (event) => {
-    event.preventDefault(true);
 
-    let res = await fetch("/api/users/register", {
-      body: JSON.stringify({
-        firstName: event.target.firstName.value,
-        lastName: event.target.lastName.value,
-        email: event.target.email.value,
-        password: event.target.password.value,
-      }),
-      headers: {
-        "Content-Type": "application/json",
-      },
-      method: "POST",
-    });
-    res = await res.json();
-    if (!res.success) {
-      return;
+  /**
+   * Displays errors if sign-up form fields do not match Zod schema
+   */
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm({ resolver: zodResolver(signUpSchema) });
+
+  /**
+   * 
+   * Performs credential sign-up on sign-up form submit
+   * Param data is the form field values
+   */
+  const onSubmit = (data) => {
```

Comment:
you may have to include `e.preventDefault()` somewhere in the submit handler since I am getting the default input error handling behavior and not our custom errors:
<img width="498" alt="Screenshot 2024-02-16 at 11 10 36 PM" src="https://github.com/GTBitsOfGood/canine-assistants/assets/45372449/ee8f8f89-f84f-41e1-adb3-3a134dfbab38">